src/utilities/updateBrowserUrl.ts

```diff
@@ -0,0 +1,39 @@
+import { isBrowser } from '@/utilities/isBrowser'
+
+type UpdateBrowserUrlOptions = {
+  replace?: boolean,
+}
+
+export function updateBrowserUrl(url: string, options: UpdateBrowserUrlOptions = {}): void {
```

Comment:
on a popstate event where the browser is telling us the user navigated outside of our router's control, would you imagine we call this function?

Reply:
if a popstate event is emitted the browser already has an updated url. So we wouldn't need to call this.